src/deye_set_time_processor.py

```diff
@@ -40,12 +39,9 @@ def get_id(self):
     def last_status(self):
         return self.__last_status
 
-    def process(self, events: list[DeyeEvent]):
-        logger_status_events: list[DeyeLoggerStatusEvent] = [
-            event for event in events if isinstance(event, DeyeLoggerStatusEvent)
-        ]
-        if logger_status_events:
-            logger_status = logger_status_events[0].online
+    def process(self, events: DeyeEventList):
+        logger_status = events.get_status_event()
+        if logger_status is not None:
             if not self.__last_status and logger_status:
```

Comment:
You've changed the type of `logger_status` from bool to `DeyeLoggerStatusEvent`. So you should update the condition too.

Reply:
Looking at this again, I think it makes more sense to change the method `get_status_event` to return the event value (or None) directly. Will change and rename method to `get_status`.

Reply:
done